common/types.go

```diff
@@ -186,6 +186,42 @@ func (h Hash) Value() (driver.Value, error) {
 	return h[:], nil
 }
 
+// GetLastNumBitsFromHash returns the last num bits of the hash as int64
+func (h Hash) GetLastNumBitsFromHash(num int) int64 {
```

Comment:
The bits to bytes to ints arrays/slice deconstruction is probably not very efficient. I would recommend an integer math approach like:
```
func (h Hash) GetTrailingBits(n int) -> uint64 {
    bytes := n / 256 // number of whole bytes to keep
    bits := n % 256  // number of remaining MSBs to keep
    bits_mask := (1<<bits)-1

    composite := h[bytes] & bits_mask // initialize with MSBs
    for i := bytes-1; i >= 0; i-- {
        composite *= 256
        composite += h[i]
    }
    return composite
}
```

Reply:
THanks Boyd, but there are only 8 bits in a byte ;). Glad I could help.